### PR TITLE
Pokedex assigment

### PR DESCRIPTION
**Pull request by OfirSimhi1612**

### CODE REVIEW

File: Main.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+// const fetchPokemon = async (id) => {
+//   try {
+//     const pokData = await axios.get(`https://pokeapi.co/api/v2/pokemon/${id}/`);
+//     return pokData.data;
+//   } catch (e) {
+//     console.log(e);
+//   }
+// };
+
+const fetchPokemon = (id) => {
```
Comment (Cyber4sPopo) [inline Main.js:10]: use regular function syntax for to level defined functions

File: Main.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+   return fetch(`https://pokeapi.co/api/v2/pokemon/${id}/`)
+  .then(res => res.json())
+  .then(data => data)
+  .catch(e => console.log(e));
+  
+};
+
+const findAllType = async (link) => {
```
Comment (Cyber4sPopo) [inline Main.js:19]: `findAllType` should return the result and then passed to another function that inserts to the DOM.
Its better to separate network logic from dom logic.
https://en.wikipedia.org/wiki/Separation_of_concerns

File: Main.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      pok.innerText = element.pokemon.name;
+      pokList.appendChild(pok);
+      pok.addEventListener("click", () => displayPokemon(pok.innerText));
+    });
+    pokList.hidden = true;
+    link.appendChild(pokList);
+  }
+
+  function hide(e){
+      if(e.target.closest('ul').className === 'list'){
```
Comment (Cyber4sPopo) [inline Main.js:34]: indentation

File: Main.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+          return;
+      }
+    let list = e.target.querySelector('.list');
+    // debugger;
+    if(list.hidden){
```
Comment (Cyber4sPopo) [inline Main.js:39]: `list.hidden = !list.hidden`

File: Main.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+   return fetch(`https://pokeapi.co/api/v2/pokemon/${id}/`)
+  .then(res => res.json())
+  .then(data => data)
+  .catch(e => console.log(e));
+  
+};
+
+const findAllType = async (link) => {
+    let typeData = await axios.get(link.getAttribute("pokType"));
```
Comment (Cyber4sPopo) [inline Main.js:20]: try using XHR or awaiting on a fetch for practice

File: Main.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
+// const fetchPokemon = async (id) => {
```
[the remark this answers is not in the document]
Comment (OfirSimhi1612) [reply Main.js:1]: that's the axios replacement for the fetch methos taking place in the the FetchPokemon function.

File: Main.js (hunk 1 above)
Comment (OfirSimhi1612) [reply Main.js:10]: done

File: Main.js (hunk 2 above)
Comment (OfirSimhi1612) [reply Main.js:19]: done

File: Main.js (hunk 3 above)
Comment (OfirSimhi1612) [reply Main.js:34]: done

File: Main.js (hunk 4 above)
Comment (OfirSimhi1612) [reply Main.js:39]: done

**Pull request merged**